src/Query/Closure/Value/VariableValue.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Bdf\Prime\Query\Closure\Value;
+
+use ReflectionFunction;
+
+/**
+ * Get value from bound variable
+ */
+final class VariableValue implements ComparisonValueInterface
+{
+    private string $variable;
+
+    /**
+     * @param string $variable The variable name
+     */
+    public function __construct(string $variable)
+    {
+        $this->variable = $variable;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get(ReflectionFunction $reflection)
+    {
+        if ($this->variable === 'this') {
+            return $reflection->getClosureThis();
+        }
+
+        return $reflection->getStaticVariables()[$this->variable];
+    }
```

Comment:
Could we manage expression that use external variable ?
`$id = 123;`
`$query->filter(fn (Entity $entity) => $entity->id() === $id);`

Reply:
This is already handled by getStaticVariables call : https://github.com/b2pweb/bdf-prime/blob/636d91228a1a1b6611af0b619779c85ef17e37d8/tests/Query/Closure/ClosureCompilerTest.php#L39